[TAN-688] Feature flag to disable user avatars

adessy opened the pull request:
Back-end part of TAN-688.

adessy committed:
[TAN-668] Remove dead code

adessy committed:
[TAN-668] Implement the user_avatars feature flag

The feature flag can be used to enable/disable the usage of avatar/profile pictures for users.

adessy committed:
[TAN-668] Rework parameter extraction for UsersController#update

adessy committed:
[TAN-668] Rework CustomFieldService#cleanup_custom_field_values!

adessy committed:
[TAN-668] Simplify permitted attributes in user_policy.rb

adessy commented:
@brentguf Do you prefer to implement the front-end changes in this branch or in your own branch? In the latter case, I'll just merge that one without a changelog entry.

brentguf commented:
Working on the FE in a different branch. I'll ping you when I'm done.

FYI, when I use my CL email (Google) to create an account on this branch, with user_avatars turned off (enabled is false in database) it still grabs my Google picture.

brentguf committed:
Hide avatar uploader when user_avatars is disabled

adessy commented:
> FYI, when I use my CL email (Google) to create an account on this branch, with user_avatars turned off (enabled is false in database) it still grabs my Google picture.

If you do have time, could you try in a private window? I observed a similar issue, but I attributed it to some caching mechanism.

brentguf commented:
> > FYI, when I use my CL email (Google) to create an account on this branch, with user_avatars turned off (enabled is false in database) it still grabs my Google picture.
> 
> If you do have time, could you try in a private window? I observed a similar issue, but I attributed it to some caching mechanism.

Just replicated it in a private window (to be clear: removed account, created again, still had the Google picture as avatar)

brentguf committed:
Merge pull request #6816 from CitizenLabDotCo/tan-688/front-end

Hide avatar uploader when user_avatars is disabled

brentguf commented:
Merged the FE.

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into tan-688/disable-user-avatars

adessy committed:
[TAN-668] Fix app_configuration caching issue

We cannot memoize app configuration because we reuse the same OmniauthMethods::Base instances over time and across different tenants.

adessy commented:
It should working now. Thanks for flagging @brentguf

brentguf commented:
@adessy yep, works as expected now!

jamesspeake commented on back/app/services/custom_field_service.rb:
```diff
@@ -88,15 +88,15 @@ def generate_token(str)
+  # @example
+  #  compact_custom_field_values!({a: 1, b: '', c: false, d: nil})
+  #  # => {a: 1, c: false}
+  def compact_custom_field_values!(cf_values)
```
A matter of taste maybe but I'm not a big fan of abbreviations in variable names. Probably fairly clear in this instance I guess though.

jamesspeake reviewed (approved):
All good, some nice refactoring in here too. Just one minor style comment, but it's probably personal preference
